Celeste.Mod.mm/Mod/Everest/Everest.Relinker.cs

```diff
@@ -334,46 +347,72 @@ public static class Relinker {
                     Modder.ReaderParameters.SymbolStream?.Dispose();
-                    if (SharedModder) {
-                        Modder.ClearCaches(moduleSpecific: true);
+
+                    if (module != Modder.Module)
                         Modder.Module?.Dispose();
-                        Modder.Module = null;
+                    Modder.Module = null;
 
+                    if (SharedModder) {
+                        Modder.ClearCaches(moduleSpecific: true);
                     } else {
                         Modder.Dispose();
                         Modder = null;
                     }
                 }
 
-                if (File.Exists(cachedChecksumPath)) {
-                    File.Delete(cachedChecksumPath);
-                }
-                if (!temporaryASM) {
-                    File.WriteAllLines(cachedChecksumPath, checksums);
-                }
-
-                Logger.Log(LogLevel.Verbose, "relinker", $"Loading assembly for {meta} - {asmname}");
                 try {
-                    Assembly asm = Assembly.LoadFrom(cachedPath);
-                    _RelinkedAssemblies.Add(asm);
-                    return asm;
-                } catch (Exception e) {
-                    Logger.Log(LogLevel.Warn, "relinker", $"Failed loading {meta} - {asmname}");
-                    e.LogDetailed();
-                    return null;
+                    if (File.Exists(cachedChecksumPath)) {
+                        File.Delete(cachedChecksumPath);
+                    }
+                    if (!temporaryASM) {
+                        File.WriteAllLines(cachedChecksumPath, checksums);
+                    }
+
+                    Logger.Log(LogLevel.Verbose, "relinker", $"Loading assembly for {meta} - {asmname} - {module.Assembly.Name}");
+                    if (Modder != null)
+                        foreach(AssemblyNameReference aref in module.AssemblyReferences) {
+                            if (Modder.DependencyCache.ContainsKey(aref.FullName))
+                                Logger.Log(LogLevel.Verbose, "relinker", $"dep. {module.Name} -> (({aref.FullName}), ({aref.Name})) found");
+                            else
+                                Logger.Log(LogLevel.Warn, "relinker", $"dep. {module.Name} -> (({aref.FullName}), ({aref.Name})) NOT FOUND");
+                        }
+
+                    try {
+                        Assembly asm = Assembly.LoadFrom(cachedPath);
+                        _RelinkedAssemblies.Add(asm);
+
+                        if (!_RelinkedModules.ContainsKey(module.Assembly.Name.Name)) {
+                            _RelinkedModules.Add(module.Assembly.Name.Name, module);
+                            module = null;
+                        } else
+                            Logger.Log(LogLevel.Warn, "relinker", $"Encountered module name conflict loading assembly {meta} - {asmname} - {module.Assembly.Name}");
```

Comment:
Would it be possible to log the name of the conflicting mod?

Reply:
The `{meta}`-part of the log statement should already contain the name